src/lib.rs

```diff
@@ -132,7 +160,12 @@ pub async fn run(config: Config) -> Result<()> {
     };
 
     if let Some(docker) = &rt.docker {
-        let pull_opts = PullOpts::builder().image(repo).tag(tag).build();
+        let auth = RegistryAuth::builder()
+            .username(config.username)
```

Comment:
Should we check whether username/password is not nil before passing it to the builder? I guess since the e2e passed, this is fine?

Reply:
We could check but the `get_args()` function makes sure that this defaults to ""/"" for username/password. This doesn't break anything for public repos, at least not in my testing. I tried the `if let (Some(username), Some(password)){` pattern you had in the issue but I was getting lifecycling errors trying to add the auth.

Reply:
ok, sounds good